dbusmock/templates/networkmanager.py

```diff
@@ -287,4 +473,126 @@ def AddWiFiConnection(self, dev_path, connection_name, ssid_name, key_mgmt):
 
     main_connections.append(connection_path)
     settings_obj.Set(SETTINGS_IFACE, 'Connections', main_connections)
+
+    settings_obj.EmitSignal(SETTINGS_IFACE, 'NewConnection', 'o', [ap_path])
+
     return connection_path
+
+
+@dbus.service.method(MOCK_IFACE,
+                     in_signature='assssu', out_signature='s')
+def AddActiveConnection(self, devices, connection_device, specific_object, name, state):
+    '''Add an active connection to an existing WiFi device.
+
+    You have to a list of the involved WiFi devices, the connection path,
+    the access point path, ActiveConnection object name and connection
+    state.
+
+    Please note that this does not set any global properties.
+
+    Returns the new object path.
+    '''
+
+    conn_obj = dbusmock.get_object(connection_device)
+    settings = conn_obj.Get(CSETTINGS_IFACE, 'Settings');
+    conn_uuid = settings['connection']['uuid']
+
+    device_objects = [dbus.ObjectPath(dev) for dev in devices]
+
+    active_connection_path = '/org/freedesktop/NetworkManager/ActiveConnection/' + name
+    self.AddObject(active_connection_path,
+                   ACTIVE_CONNECTION_IFACE,
+                   {
+                       'Devices': device_objects,
+                       'Default6': False,
+                       'Default': True,
+                       'Vpn': False,
+                       'Connection': dbus.ObjectPath(connection_device),
+                       'Master': dbus.ObjectPath('/'),
+                       'SpecificObject': dbus.ObjectPath(specific_object),
+                       'Uuid': conn_uuid,
+                       'State': state,
+                   },
+                   [])
+
+    for dev_path in devices:
+        self.SetDeviceActive(dev_path, active_connection_path)
+
+    active_connections = self.Get(MAIN_IFACE, 'ActiveConnections')
+    active_connections.append(dbus.ObjectPath(active_connection_path))
+    self.SetProperty(MAIN_OBJ, MAIN_IFACE, 'ActiveConnections', active_connections)
```

Comment:
Could NetworkManager emit signal org.freedesktop.NetworkManager.PropertiesChanged with the changed props, i.e. ActiveConnections, here?